classes/Adapter/LanguageAdapter.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+* 2007-2019 PrestaShop
+*
+* NOTICE OF LICENSE
+*
+* This source file is subject to the Academic Free License (AFL 3.0)
+* that is bundled with this package in the file LICENSE.txt.
+* It is also available through the world-wide-web at this URL:
+* http://opensource.org/licenses/afl-3.0.php
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author    PrestaShop SA <contact@prestashop.com>
+*  @copyright 2007-2019 PrestaShop SA
+*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+namespace PrestaShop\Module\PrestashopCheckout\Adapter;
+
+use PrestaShop\Module\PrestashopCheckout\ShopContext;
+
+/**
+ * Language adapter
+ */
+class LanguageAdapter
+{
+    /**
+     * Language object
+     *
+     * @var \Language
+     */
+    private $language;
+
+    public function __construct(\Language $language = null)
+    {
+        if (null === $language) {
+            $language = new \Language();
+        }
+
+        $this->language = $language;
```

Comment:
Looks like this property can be removed

Reply:
Actually is seems the whole constructor may be removed. We have a private property and there is no use of `$this->language` in this class.

And as we deal with a new class, there is no backward compatibility to take care about.